WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -201,14 +199,22 @@ extension PointOfSaleAggregateModel {
         }
     }
 
-    func startCashPayment() {
-        isCashPaymentInProgress = true
+    // Prevents card payments from the moment the merchant decides we start collecting cash
+    // Once we get the callback from the card service, we switch to cash collection state
+    @MainActor
+    func startCashPayment() async {
+        try? await cardPresentPaymentService.cancelPayment()
         paymentState = .cash(.collectingCash)
     }
 
-    func cancelCashPayment() {
-        isCashPaymentInProgress = false
-        paymentState = .card(.idle)
+    @MainActor
+    func cancelCashPayment() async {
+        if case .connected = cardReaderConnectionStatus {
+            try? await cardPresentPaymentService.cancelPayment()
```

Comment:
Why do we need to cancel the card payment again here?

Reply:
We don't! It would be redundant. I think might be there from the previous attempt where we were not yet cancelling on starting cash collection. Removed: https://github.com/woocommerce/woocommerce-ios/pull/14795/commits/6e4d54ff84ac04dc7684eab447a3635d2561d164